src/solver_opts.jl

```diff
@@ -156,7 +156,7 @@ end
 
 function reset!(stats::SolverStats, N::Int=0)
     stats.is_reset && return nothing
-    stats.iterations = 0
+    stats.iterations = 1
```

Comment:
Why did you change this? This should be at zero so that the iteration counts work correctly.

Reply:
Yeah I wasn't 100% sure about the best way to fix this, but stats.iteration is used as an array index in multiple places so you get indexing errors when it starts at 0.

Reply:
Have you run into this in practice? It shouldn't ever try to index into the `stats` if it hasn't completed an iteration. I'm more inclined to find where it's trying to index when it shouldn't be, since that's a problem. Changing this won't fix the underlying issue, just prevent an error (basically making the bug harder to detect).

Reply:
yes see https://github.com/RoboticExplorationLab/Altro.jl/issues/4